### DISCUSSION TITLE
Server start  !!!

### DISCUSSION THREAD

**Discussion by sefaa-dev** [dotnet/MQTTnet#1840] [General]
what's my wrong with my code ?

`            var options = new MqttServerOptionsBuilder();
            var mqttFactory = new MqttFactory();
            mqttServer = mqttFactory.CreateMqttServer(options.WithDefaultEndpointPort(1800).Build());
`

C:\Users\DELL_05>netstat -na | find "1800"

C:\Users\DELL_05>

**Comment by CZEMacLeod** [comment]
Almost in your issue title: You _created_ your server, but didn't _start_ it.

See line 82 below.

https://github.com/dotnet/MQTTnet/blob/6fdee9e0db738817c6f457c3597f84460cff3cd0/Samples/Server/Server_Simple_Samples.cs#L61-L90

Also remember to `Dispose` your server when you are finished with it. (Implicit with the `using` in the code example above.)

**Comment by sefaa-dev** [comment]
Actually I started it but it doesn't work

`
 static void Main(string[] args)
 {
     var optionBuilder = new MqttServerOptionsBuilder()
         .WithDefaultEndpointPort(1800)
         .WithDefaultCommunicationTimeout(TimeSpan.FromMilliseconds(30*1000));
     var option = optionBuilder.Build();

     using (var mqttServer = new MqttFactory().CreateMqttServer(option))
     {
         mqttServer.StartAsync().Wait();

         Console.WriteLine("Press Enter to exit.");
         Console.ReadLine();
         mqttServer.StopAsync().Wait();
     }
 }
`

**Comment by CZEMacLeod** [comment]
@sefaa-dev You need to call the `WithDefaultEndpoint()` method as well, to actually enable the default endpoint in addition to configuring the ipaddress and/or port (actually -  both of these are technically optional as they default to 0.0.0.0:1883)

```cs
using MQTTnet;
using MQTTnet.Server;

public class Program
{
    static async Task Main(string[] args)
    {
        var logger = new MQTTnet.Diagnostics.MqttNetEventLogger();
        logger.LogMessagePublished += Logger_LogMessagePublished;
        var optionBuilder = new MqttServerOptionsBuilder()
            .WithDefaultEndpoint()
            .WithDefaultEndpointBoundIPAddress(System.Net.IPAddress.Loopback)
            .WithDefaultEndpointPort(18883);
        var option = optionBuilder.Build();
        var factory = new MqttFactory(logger);
        using var mqttServer = factory.CreateMqttServer(option);
        await mqttServer.StartAsync();

        Console.WriteLine("Press Enter to exit.");
        Console.ReadLine();
        await mqttServer.StopAsync();
    }

    private static void Logger_LogMessagePublished(object? sender, MQTTnet.Diagnostics.MqttNetLogMessagePublishedEventArgs e)
    {
        Console.WriteLine(e.LogMessage);
    }
}
```

```script
netstat -na | find "883"
```
